### PR TITLE
bpo-32494: Add gdbm.count()

### PR DESCRIPTION
**Pull request by ZackerySpytz**
gdbm_count() was added in GDBM 1.11.

https://bugs.python.org/issue32494

### CODE REVIEW

File: Modules/_gdbmmodule.c [hunk 1]
```diff
@@ -486,6 +486,32 @@ _gdbm_gdbm_sync_impl(dbmobject *self)
...
+Count the number of records in the database.
+[clinic start generated code]*/
+
+static PyObject *
+_gdbm_gdbm_count_impl(dbmobject *self)
+/*[clinic end generated code: output=8b62b02ae4c2fcee input=1b9d3f0877315a04]*/
+{
+    gdbm_count_t count;
+    check_dbmobject_open(self);
+    if (gdbm_count(self->di_dbm, &count) == -1) {
```
Comment (pitrou) [inline Modules/_gdbmmodule.c:502]: Need to set `errno = 0` before this.

**Review by pitrou — CHANGES_REQUESTED** [review]
Thank you. This looks mostly good, just one comment.

**Review by corona10 — COMMENTED** [review]
@ZackerySpytz  @pitrou 

Can we just replace the logic of `dbm_length` to use  `gdbm_count` when the version is available?

https://github.com/python/cpython/blob/1b55b65638254aa78b005fbf0b71fb02499f1852/Modules/_gdbmmodule.c#L98

if possible, we don't have to export a new method.

**Comment by pitrou** [comment]
Superseded by PR #19814. Closing.

**Pull request closed without merging**